Python/optimizer.c

```diff
@@ -591,7 +592,12 @@ translate_bytecode_to_trace(
 
     for (;;) {
         target = INSTR_IP(instr, code);
-
+        if (is_noopt) {
+            max_length-=2;
+        }
+        else {
+            max_length--;
+        }
```

Comment:
I think this is fine for now, but we should keep in mind that this means we get different unoptimized traces produced!

E.g, when it's noopt, the max_length might be a trace of 300
when it's opt, the max_length might be a trace of 600

This makes things harder to debug.

Could you try increasing `UOP_MAX_TRACE_LENGTH` to 1200.
Then keep this as `max_length -= 2;` without the condition?